lms/djangoapps/completion/tests/test_models.py

```diff
@@ -142,3 +142,30 @@ def test_cannot_call_submit_completion(self):
                 completion=0.9,
             )
         self.assertEqual(models.BlockCompletion.objects.count(), 1)
+
+
+class SubmitBatchCompletionTestCase(TestCase):
+    """
+    Test that BlockCompletion.objects.submit_batch_completion has the desired
+    semantics.
+    """
+    def setUp(self):
+        super(SubmitBatchCompletionTestCase, self).setUp()
+        _overrider = waffle.waffle().override(waffle.ENABLE_COMPLETION_TRACKING, True)
+        _overrider.__enter__()
+        self.addCleanup(_overrider.__exit__, None, None, None)
+
+        blocks = [
+            (UsageKey.from_string('block-v1:edx+test+run+type@video+block@doggos'), 1.0),
+            (UsageKey.from_string('block-v1:edx+test+run+type@video+block@doggos'), 1.0),
+            (UsageKey.from_string('block-v1:edx+test+run+type@video+block@doggos'), 0.0)
+        ]
+        self.course_key = 'course-v1:edx+test+run'
+        self.course_key_obj = CourseKey.from_string(self.course_key)
+        self.blocks = blocks
+        self.user = UserFactory()
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course_key)
+
+    def test_batch_of_three_blocks(self):
+        models.BlockCompletion.objects.submit_batch_completion(self.user, self.course_key_obj, self.blocks)
+        self.assertEqual(models.BlockCompletion.objects.count(), 3)
```

Comment:
Should also test that the completion value for the final `BlockCompletion` object matches 0.0, since that was the last completion to be sent for `block-v1:edx+test+run+type@video+block@doggo`.